src/core/__tests__/template-host.test.tsx

```diff
@@ -0,0 +1,28 @@
+import { wrapTemplate } from "../template-host";
+
+describe("template-host", () => {
+
+    describe("its wrapped template", () => {
+
+        it("doesn't render twice into the same container", () => {
+            const updateFunc = jest.fn();
+            const container: any = {};
+            const template = wrapTemplate(
+                jest.fn(),
+                updateFunc,
+                jest.fn()
+            );
+
+            template.render({
+                container
+            });
+
```

Comment:
How can I be sure that the test is correct? What if I try to clear the `container` content before the second rendering? Maybe it's better to make a test based on the real issue?

Reply:
> What if I try to clear the `container` content before the second rendering?
Actually, it would be OK, I think.

My intention was to write a more union-like test, describing how the entity called `template-host` should behave. Without React, DOM and others. So to be sure the test is correct we need to clearly understand what template-host is, what responsibilities it has and so on - then we'll test the entity, not the code.